### PR TITLE
CONTRIBUTING.md: updates for Dart 2; move parts to flutter/engine

### PR DESCRIPTION
**Pull request by yjbanov**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -239,30 +239,9 @@ the following steps.
...
-   Host configuration provides standalone dart sdk for the engine, that is used
-   to run engine dart scripts on the host.
-
-You should now be able to run the tests against your locally built
-engine using the `flutter test --local-engine=host_debug_unopt` command. To run
-one of the examples on your device using your locally built engine, use the
-`--local-engine=android_debug_unopt` option to the `flutter` tool:
-
- * `flutter run --local-engine=android_debug_unopt`
+   for the engine repository to create a working copy of the engine. The instructions
```
Comment (cbracken) [inline CONTRIBUTING.md]: s/for the/in the/

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -239,30 +239,9 @@ the following steps.
...
+   also explain how to use the locally built engine instead of the one bundled with
```
Comment (cbracken) [inline CONTRIBUTING.md]: nit: s/the locally built/a locally-built/

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm

File: CONTRIBUTING.md (hunk 1 above)
Comment (yjbanov) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 2 above)
Comment (yjbanov) [reply CONTRIBUTING.md]: Done.

**Comment by yjbanov** [comment]
Submitting on red build as this change is harmless.

**Pull request merged**